src/graphql/assets.ts

```diff
@@ -1,62 +1,37 @@
-import type {
-  ContentTypeProps,
-  EntryProps,
-  AssetProps,
-  ContentFields,
-} from 'contentful-management/types';
+import type { AssetProps, EntryProps } from 'contentful-management/types';
 
-import { Entity, EntryReferenceMap, SysProps } from '../types';
-import { updateEntry } from './entries';
-
-const field = (name: string, type = 'Symbol'): ContentFields => ({
-  id: name,
-  name,
-  type,
-  localized: true,
-  required: false,
-});
-
-/**
- * Hand-coded Asset Content Type to have consistent handling for Entries
- * and Assets.
- */
-const AssetContentType = {
-  name: 'Asset',
-  fields: [
-    field('title'),
-    field('description'),
-
-    // File attributes
-    field('fileName'),
-    field('contentType'),
-    field('url'),
-    field('size', 'Integer'),
-    field('width', 'Integer'),
-    field('height', 'Integer'),
-  ],
-} as ContentTypeProps;
+import { Entity, SysProps } from '../types';
 
 /**
  * Updates GraphQL response data based on CMA Asset object
  *
- *
- * @param data the GraphQL response to be updated
- * @param asset CMA Asset object containing the update
+ * @param dataFromPreviewApp the GraphQL response to be updated
+ * @param updateFromEntryEditor CMA Asset object containing the update
  * @param locale locale code
  */
-export function updateAsset(
-  data: Entity & { sys: SysProps },
-  update: AssetProps,
-  locale: string,
-  entityReferenceMap: EntryReferenceMap
-): Entity {
-  // FIXME: copy nested asset.fields.file values to root to match the
-  // Content Type definition for GraphQL
-  return updateEntry(
-    AssetContentType,
-    data,
-    update as unknown as EntryProps,
-    locale,
-    entityReferenceMap
-  );
+export function updateAsset<T extends (Entity & { sys: SysProps }) | EntryProps>(
+  dataFromPreviewApp: T,
+  updateFromEntryEditor: AssetProps,
+  locale: string
+): T {
+  try {
+    const file = updateFromEntryEditor.fields.file[locale];
+
+    // Content Type definition for GraphQL
+    return {
+      ...dataFromPreviewApp,
+      // GraphQL flattens some information
+      // and as the live updates are coming from the CMA, we need to transform them
+      title: updateFromEntryEditor.fields.title[locale],
+      description: updateFromEntryEditor.fields.description?.[locale],
+      contentType: file.contentType,
+      width: file.details?.image.width,
+      height: file.details?.image.height,
+      // GraphQL returns the URL with protocol and the CMA without it, so we need to add the information in there manually
+      url: file.url ? (file.url?.startsWith('https:') ? file.url : `https:${file.url}`) : undefined,
```

Comment:
Could maybe extract this to a util function

```
function updateProtocolToHttps(urlString: string | undefined): string | undefined {
  if (typeof urlString !== 'string') {
    return urlString;
  }

  try {
    const url = new URL(urlString);
    url.protocol = 'https:';
    return url.href;
  } catch (error) {
    console.error(`Error updating protocol to https: ${error}`);
    return urlString;
  }
}
```

Reply:
Needed to adjust it, as `new URL('//images.ctfassets.net/')` is invalid, it needs the protocol :(